subworkflows/local/input_check.nf

```diff
@@ -7,33 +7,28 @@ include { SAMPLESHEET_CHECK } from '../../modules/local/samplesheet_check'
 workflow INPUT_CHECK {
     take:
     complete_samplesheet // file: /path/to/complete_samplesheet.csv
-    data_dir
-    
+
     main:
-    SAMPLESHEET_CHECK ( complete_samplesheet, data_dir )
-    
+    SAMPLESHEET_CHECK(complete_samplesheet)
+
     SAMPLESHEET_CHECK.out
-        .complete_csv
-        .splitCsv ( header:true, sep:',' )
+        .csv
+        .splitCsv (header:true, sep:',')
         .map { create_prots_channel(it) }
         .set { complete_prots }
-        
-    SAMPLESHEET_CHECK.out
-        .mcl_test_csv
-        .splitCsv ( header:true, sep:',' )
-        .map { create_prots_channel(it) }
-        .set { mcl_test_prots }
-    
+
```

Comment:
Okay, so this negates the need for all of my absurd bash scripting, just using nextflow operators to parse the samplesheet into two channels, one with all samples, one with the mcl_test == 'true' condition? Which in turn should inherently keep analysis of the two separate (i.e. staging their files independently)?

Beautiful